test.lisp

```diff
@@ -1,22 +1,87 @@
 (in-package :cl)
 
 (defpackage :inferior-shell-test
-  (:use :cl :inferior-shell :hu.dwim.stefil))
+  (:use :cl :inferior-shell :fiveam))
 
 (in-package :inferior-shell-test)
 
 (declaim (optimize (speed 1) (debug 3) (space 3)))
 
-(defsuite* (test-suite
-            :in root-suite
-            :documentation "Testing inferior-shell"))
+(defmacro with-temp-file (symbol &rest body)
+  `(let ((,symbol (format nil "/tmp/inf-shell-file-~a" (random 100000))))
+     (run/ss `(rm ,,symbol) :on-error nil)
+     (unwind-protect
+          (progn ,@body)
+       (run/ss `(rm ,,symbol) :on-error nil))))
 
-(defun do-test-inferior-shell ()
-  (is (equal (run/ss '(echo (1) "2" (+ 3))) "1 2 3"))
-  (is (equal (run/ss "echo 1    2        3") "1 2 3"))
-  (is (equal (run/ss `(pipe (echo (+ hel "lo,") world)
-                            (tr "hw" "HW") (sed -e "s/$/!/")))
-             "Hello, World!")))
+(def-suite inferior-shell-test
+  :description "Suite of tests for inferior shell.")
 
-(deftest test-inferior-shell ()
```

Comment:
Why the indirection?

Reply:
I have more tests now, so I break them into sub-suites.